File: frontend/src/services/enrollmentService.ts
import apiClient from "./apiClient";

export interface EnrollmentResponse {
  id: string;
  userId: string;
  courseId: string;
  courseTitle: string;
  courseThumbnailUrl?: string;
  coursePrice: number;
  progress: number;
  enrolledAt: string;
}

const enrollmentService = {
  enrollCourse: async (courseId: string): Promise<EnrollmentResponse> => {
    const res = await apiClient.post(`/courses/${courseId}/enroll`);
    return res.data.data;
  },

  getMyCourses: async (): Promise<EnrollmentResponse[]> => {
    const res = await apiClient.get("/me/courses");
    return res.data.data;
  },

  updateProgress: async (courseId: string, progress: number): Promise<EnrollmentResponse> => {
    const res = await apiClient.put(`/me/courses/${courseId}/progress`, null, {
      params: { progress },
    });
    return res.data.data;
  },
};

export default enrollmentService;

File: backend/src/main/java/com/study4you/course/service/EnrollmentService.java
package com.study4you.course.service;

import com.study4you.course.dto.EnrollmentResponse;

import java.util.List;
import java.util.UUID;

public interface EnrollmentService {

    EnrollmentResponse enrollCourse(UUID courseId, UUID userId);

    List<EnrollmentResponse> getMyCourses(UUID userId);

    EnrollmentResponse updateProgress(UUID courseId, UUID userId, int progress);

    boolean isEnrolled(UUID courseId, UUID userId);
}
